libsolidity/codegen/ir/Common.cpp

```diff
@@ -127,9 +127,10 @@ string IRNames::zeroValue(Type const& _type, string const& _variableName)
 	return "zero_" + _type.identifier() + _variableName;
 }
 
-string sourceLocationComment(langutil::SourceLocation const& _location, IRGenerationContext const& _context)
+string sourceLocationComment(langutil::SourceLocation const& _location, IRGenerationContext& _context)
 {
 	solAssert(_location.sourceName, "");
+	_context.markSourceUsed(*_location.sourceName);
```

Comment:
This might need a better name. I'd not expect side-effects from it.

Reply:
I don't understand - `mark` is an action, so it should have side-effects. Can you suggest a name?

Reply:
Sorry I meant `sourceLocationComment()` but commented on a bigger fragment to make the context visible.

How about `dispenseLocationComment()`? It would suggest that getting this comment updates some internal state and that you should expect some side-effects.

Reply:
I like that name suggestion